EIPS/eip-3756.md

```diff
@@ -0,0 +1,55 @@
+---
+eip: 3756
+title: Gas Limit Cap
+description: Set an in-protocol cap for the gas limit
+author: lightclient (@lightclient)
+discussions-to: https://ethereum-magicians.org/t/eip-3756-gas-limit-cap/6921
+status: Draft
+type: Standards Track
+category: Core
+created: 2021-08-21
+---
+
+## Abstract
+
+Set an in-protocol cap for the gas limit of 30,000,000.
+
+## Motivation
+
+A high gas limit increases pressure on the network. In the benign case, it
+increases the size of the state and history faster than we can sustain. In the
+malicious case, it amplifies the devestation of certain denial-of-service
+attacks.
+
+## Specification
+
+As of the fork block `N`, consider blocks with a `gas_limit` greater than
```

Comment:
It may be valuable to be more explicit about what `exactly` is invalid here.  In this case it is the 10th RLP encoded item in the block header that must be a QUANTITY that is less than or equal to 30,000,000.

Reply:
Counting RLP fields is too pedantic for readability.  I think referencing EIP-1559 in the header and then "a `block` with `gas_limit` greater than `30,000,000`" reads much better.